viewing_party/party.py

```diff
@@ -1,23 +1,224 @@
+## use the new_movie list to generate how the function works
+## user_data is what has been "watched:"
+## updated_data = add_to_watched(user_data, movie)
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+## Create a function to track a watch list for the user
+## user_data = {"watchlist"}
+## updated_data = add_to_watchlist(user_data, movie)
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+## def a function that will call upon the watched and watchlist functions
+##  adjust the values of watched and watchlist into sets
+## use set functions to create a set that compares the similarities
+## return a set or dictionary that outputs the difference 
+## Return an error message if the movie is on neither list
+
+def watch_movie(user_data, title):  
+    for movies in user_data["watchlist"]:
+        if movies["title"] == title: 
+            user_data["watchlist"].remove(movies)
+            user_data["watched"].append(movies)
+    return user_data
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+## Create a function that will check the average rating of movies watched
+## if average is Zero, return error message
+
+'''
+input: a dictionary called 'user_data' with a 'watched' list of
+movie dictionaries (movies that user has watched)
+-- Calculate average rating of all movies user has watched
+-- An empty watch list has an average rating of 0.0
+output: the average rating
+# {   'watched': [   {   'genre': 'Fantasy',
+#                        'rating': 4.8,
+#                        'title': 'The Lord of the Functions: The Fellowship of '
+#                                 'the Function'},
+#                    {   'genre': 'Fantasy',
+#                        'rating': 4.0,
+#                        'title': 'The Lord of the Functions: The Two '
+#                                 'Parameters'},
+#                    {   'genre': 'Fantasy',
+#                        'rating': 4.0,
+#                        'title': 'The Lord of the Functions: The Return of the '
+#                                 'Value'},
+#                    {   'genre': 'Action',
+#                        'rating': 2.2,
+#                        'title': 'The JavaScript and the React'},
+#                    {'genre': 'Intrigue', 'rating': 2.0, 'title': 'Recursion'},
+#                    {   'genre': 'Intrigue',
+#                        'rating': 4.5,
+#                        'title': 'Instructor Student TA Manager'}]}
+'''
+
+def get_watched_avg_rating(user_data):
+    watched_movies = user_data["watched"] ## array of dictionaries
+    total_rating = []
+    counter = 0 
+## for a dictionary in the list of movies
+    for movie_data in watched_movies:
+        v = movie_data["rating"]
+        total_rating.append(v)
+    if len(total_rating) == 0:

```

Comment:
The preferred syntax for checking if a list is empty in python is to use the truthy/falsyness of the list (just like the input validation for `create_movie` above!)

```
    if not total_rating:
```